### PR TITLE
41466

### PR DESCRIPTION
**Pull request by ZASMan**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [41466](https://devops.dchbx.org/redmine/issues/41466) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | 2 |
| Peer Review Story Points | n/a |
| Ticket Author | @ap-dc |

### CODE REVIEW

File: app/controllers/employers/employer_profiles_controller.rb [hunk 1]
```diff
@@ -302,6 +302,29 @@ def employer_account_creation_notice
     end
   end
 
+  def terminate_employee_roster_enrollments
+    authorize HbxProfile, :modify_admin_tabs?
+    employer_profile = EmployerProfile.find(params["employer_profile_id"])
+    termination_date = params["termination_date"].to_date
```
Comment (raghuramg) [inline app/controllers/employers/employer_profiles_controller.rb]: move this logic into model (like employer_profile.terminate_employee_enrollments)...also use strong parameters with `permit`

File: app/views/employers/employer_profiles/my_account/_census_employees.html.erb [hunk 2]
```diff
@@ -51,7 +51,10 @@
               <%= link_to "<i class='fa fa-file-text-o' aria-hidden='true'></i> Upload Employee Roster".html_safe, new_employers_employer_profile_census_employee_path(@employer_profile, modal: 'upload'), remote: 'true', class: "btn btn-default" %>
             </span>
             <%= link_to "<i class='fa fa-cloud-download' aria-hidden='true'></i> Download Employee Roster".html_safe, employers_employer_profile_export_census_employees_path(@employer_profile, format: :csv), class: "btn btn-default" %>
-
+            <% role = current_user.person.hbx_staff_role %>
```
Comment (raghuramg) [inline app/views/employers/employer_profiles/my_account/_census_employees.html.erb]: Use pundit policy here...you can add employer_profile_policy and view condition like `policy(@employer_profile).modify_admin_tabs?`

File: app/controllers/employers/employer_profiles_controller.rb [hunk 3]
```diff
@@ -302,6 +302,29 @@ def employer_account_creation_notice
...
+    termination_reason = params["termination_reason"]
+    transmit_xml = params["transmit_xml"]
+
+    active_py = employer_profile.active_plan_year
+    renewing_py = employer_profile.renewing_plan_year
+    
+    if active_py.present?
```
Comment (raghuramg) [inline app/controllers/employers/employer_profiles_controller.rb]: What if active plan year missing....it still show success message. Please handle the scenario where active plan year missing

File: app/controllers/employers/employer_profiles_controller.rb (hunk 1 above)
Comment (ZASMan) [reply app/controllers/employers/employer_profiles_controller.rb]: Acknowledged. I moved it to the model and added strong params.

File: app/views/employers/employer_profiles/my_account/_census_employees.html.erb (hunk 2 above)
Comment (ZASMan) [reply app/views/employers/employer_profiles/my_account/_census_employees.html.erb]: Acknowledged. I added a policy method/check and a spec to go with it.

File: app/controllers/employers/employer_profiles_controller.rb (hunk 3 above)
Comment (ZASMan) [reply app/controllers/employers/employer_profiles_controller.rb]: Acknowledged. I added conditionals.

**Pull request merged**